tests/components/ring/test_init.py

```diff
@@ -81,36 +92,35 @@ async def test_auth_failed_on_setup(
 )
 async def test_error_on_setup(
     hass: HomeAssistant,
-    requests_mock: requests_mock.Mocker,
+    mock_ring_client,
     mock_config_entry: MockConfigEntry,
     caplog,
     error_type,
     log_msg,
 ) -> None:
-    """Test auth failure on setup entry."""
+    """Test non-auth errors on setup entry."""
     mock_config_entry.add_to_hass(hass)
-    with patch(
-        "ring_doorbell.Ring.update_data",
-        side_effect=error_type,
-    ):
-        await hass.config_entries.async_setup(mock_config_entry.entry_id)
-        await hass.async_block_till_done()
 
-        assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+    mock_ring_client.update_data.side_effect = error_type
 
-        assert [
-            record.message
-            for record in caplog.records
-            if record.levelname == "DEBUG"
-            and record.name == "homeassistant.config_entries"
-            and log_msg in record.message
-            and DOMAIN in record.message
-        ]
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+
+    assert [
+        record.message
+        for record in caplog.records
+        if record.levelname == "DEBUG"
+        and record.name == "homeassistant.config_entries"
+        and log_msg in record.message
+        and DOMAIN in record.message
+    ]
```

Comment:
We usually dislike if statements in tests. Would it make sense to just check like `assert log_msg in caplog.text`?

Reply:
Done